scripts/gateway-docker-upgrade.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+set -e
+
+TARGET_IMAGE="${TARGET_IMAGE:-us-east1-docker.pkg.dev/firezone-prod/firezone/gateway:1}"
+REPO=$(dirname "$TARGET_IMAGE")
+IMAGE=$(basename "$TARGET_IMAGE")
+
+CURRENTLY_RUNNING=$(docker ps --format "{{.Names}} {{.Image}}" | grep -e "$TARGET_IMAGE" | awk '{print $1}')
+if [ "$CURRENTLY_RUNNING" == "" ]; then
+    echo "No Firezone gateway found running on this system. Exiting."
+    exit -1
+fi
+
+echo "Pulling latest image..."
+docker pull "$TARGET_IMAGE" > /dev/null
+echo "Checking for containers to upgrade..."
+for RUNNING_CONTAINER in $CURRENTLY_RUNNING
+do
+    LATEST=$(docker inspect --format "{{.Id}}" "$TARGET_IMAGE")
+    RUNNING=$(docker inspect --format "{{.Image}}" "$RUNNING_CONTAINER")
+    RUNNING_NAME=$(docker inspect --format "{{.Name}}" "$RUNNING_CONTAINER" | sed 's~/~~g')
+
+    # Upgrade if necessary
+    if [ "$RUNNING" != "$LATEST" ]; then
+        echo -n "Upgrading gateway..."
+        docker container inspect "$RUNNING_CONTAINER" --format '{{join .Config.Env "\n"}}' | grep -v "PATH" > variables.env
+        docker stop "$RUNNING_CONTAINER" > /dev/null
+        docker rm -f "$RUNNING_CONTAINER" > /dev/null
+        docker run -d \
```

Comment:
Let me think about it for a bit - would be bad to have one more point of maintenance for all those params, and saving token to file doesn't feel secure, even if for a brief time. Maybe I will come up with something better

Reply:
Yeah, I see what you mean. The other option would be to just have one script for both deployment and upgrade but that would be more complex. I suppose having the file there is not any worse because the user running this script could just fetch the vars from the running system anyway.

Maybe we could `umask` it at least